spec/services/access_request_approval_service_spec.rb

```diff
@@ -130,5 +130,11 @@
         end
       end
     end
+
+    context "writing data concurrently" do
+      it "writes to both user_permission and organisation_user" do
+        expect { subject }.to change { OrganisationUser.count }.by(1).and change { UserPermission.count }.by(access_request.requester.organisations.flat_map(&:providers).count)
```

Comment:
Related to the above — this only passes because the organisation doesn't have any associated providers. Need some data setup first in this context block.